memory/src/page.rs

```diff
@@ -3,7 +3,7 @@
 /// A single page of memory
 ///
 /// This type is page-aligned and page-sized.
-#[derive(Copy, Clone)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Ord, PartialOrd)]
```

Comment:
I don't understand how we can support `PartialOrd` and `Ord` in this case. Also, the default debug output isn't pretty.